packages/transport/src/modules/relationships/RelationshipsController.ts

```diff
@@ -221,299 +228,172 @@ export class RelationshipsController extends TransportController {
+            // this path is for a revocation that is processed before its corresponding creation
+            relationshipDoc = await this.relationships.read(relationshipId);
         }
 
-        const responseContent = await this.decryptCreationChangeResponse(change, relationship.relationshipSecretId);
-
-        const response = RelationshipChangeResponse.fromBackbone(change.response, responseContent.content);
-
-        if (!relationship.cache) {
-            throw this.newCacheEmptyError(Relationship, relationship.id.toString());
-        }
-        relationship.cache.changes[0].status = change.status;
-        switch (change.status) {
-            case RelationshipChangeStatus.Accepted:
-                relationship.toActive(response);
-                break;
-            case RelationshipChangeStatus.Rejected:
-                relationship.toRejected(response);
-                break;
-            case RelationshipChangeStatus.Revoked:
-                relationship.toRevoked(response);
-                break;
-            default:
-                throw CoreErrors.general.notSupported();
-        }
-
-        await this.relationships.update(relationshipDoc, relationship);
-        return relationship;
+        return await this.updatePendingRelationshipWithPeerResponse(relationshipDoc);
     }
 
-    @log()
-    private async createNewRelationshipByIncomingCreationChange(change: BackboneGetRelationshipsChangesResponse): Promise<Relationship> {
-        const backboneRelationship = (await this.client.getRelationship(change.relationshipId)).value;
+    private async prepareAcceptanceContent(relationship: Relationship) {
+        const publicAcceptanceContentCrypto = await this.secrets.getPublicAcceptanceContentCrypto(relationship.relationshipSecretId);
 
-        const templateId = CoreId.from(backboneRelationship.relationshipTemplateId);
-        const template = await this.parent.relationshipTemplates.getRelationshipTemplate(templateId);
+        const acceptanceContent = RelationshipAcceptanceContentWrapper.from({
+            relationshipId: relationship.id,
+            content: ""
+        });
 
-        if (!template) throw CoreErrors.general.recordNotFound(RelationshipTemplate, templateId.toString());
-        if (!template.cache) throw this.newCacheEmptyError(RelationshipTemplate, template.id.toString());
-        if (!change.request.content) throw this.newEmptyOrInvalidContentError(change);
+        const serializedAcceptanceContent = acceptanceContent.serialize();
+        const buffer = CoreUtil.toBuffer(serializedAcceptanceContent);
 
-        const secretId = await TransportIds.relationshipSecret.generate();
-        const requestCipher = RelationshipCreationChangeRequestCipher.fromBase64(change.request.content);
-        await this.secrets.createTemplatorSecrets(secretId, template.cache, requestCipher.publicRequestCrypto);
```

Comment:
This seems wrong, why is this removed?

Reply:
what was removed?

Reply:
I think it's only re-positioned and renamed, that's thy GH shows a big diff.